### PR TITLE
feat(dashboard): transfer silence switch chain

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
MF-3520

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx [hunk 1]
```diff
@@ -70,11 +69,11 @@ export const TransferERC20 = memo<TransferERC20Props>(({ token }) => {
...
     const [selectedToken, setSelectedToken] = useState(token)
     const selectFungibleToken = useSelectFungibleToken<void, NetworkPluginID.PLUGIN_EVM>()
-    const { chainId } = useChainContext<NetworkPluginID.PLUGIN_EVM>()
-    const is1559Supported = useMemo(() => chainResolver.isSupport(chainId, 'EIP1559'), [chainId])
 
+    const { chainId, pluginID } = useContainer(Context)
     const Others = useWeb3Others()
-
+    const is1559Supported = useMemo(() => Others?.chainResolver.isSupport(chainId, 'EIP1559'), [chainId])
+    console.log({ chainId, token })
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx]: remove log

File: packages/web3-hooks/evm/src/useNativeTokenTransferCallback.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ export function useNativeTransferCallback() {
                 value: amount,
                 data: memo ? toHex(memo) : undefined,
                 ...gasConfig,
+                chainId,
```
Comment (guanbinrui) [inline packages/web3-hooks/evm/src/useNativeTokenTransferCallback.ts:39]: It might not necessary.

Comment (hanszhoou) [reply packages/web3-hooks/evm/src/useNativeTokenTransferCallback.ts:39]: Tested, it is required here.

File: packages/dashboard/src/pages/Wallets/StartUp.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ export function StartUp() {
                             action={{
                                 type: 'primary',
                                 text: t.wallets_startup_connect_action(),
-                                handler: SelectProviderDialog.open,
+                                handler: () => SelectProviderDialog.open(),
```
Comment (hanszhoou) [inline packages/dashboard/src/pages/Wallets/StartUp.tsx:31]: `handler: SelectProviderDialog.open` not work.

Comment (UncleBill) [reply packages/dashboard/src/pages/Wallets/StartUp.tsx:31]: We can make it work. see https://github.com/DimensionDev/Maskbook/pull/9849

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**